File: src/types/doctor.ts
export interface Doctor {
  doctorId: number
  doctorName: string
  fullName?: string
  specialtyId: number
  specialtyName: string
  examFee: number
  degree?: string
  experienceYears?: number
  phone?: string
  email?: string
  gender?: string
  dateOfBirth?: string
  description?: string
  avatarUrl?: string
  roomNumber?: string
  userId?: string | number | null
  createdAt?: string
  updatedAt?: string | null
  isActive?: boolean
}

export interface DoctorSchedule {
  scheduleId: number
  doctorId: number
  doctorName: string
  workDate: string
  startTime: string
  endTime: string
  slotDurationMinutes?: number
  isAvailable?: boolean
}

File: backend/AppointmentService/Models/Doctor.cs
namespace AppointmentService.Models;

public sealed class Doctor
{
    public int Id { get; set; }

    public int? UserId { get; set; }

    public string FullName { get; set; } = string.Empty;

    public int SpecialtyId { get; set; }

    public string Degree { get; set; } = string.Empty;

    public int ExperienceYears { get; set; }

    public decimal ExamFee { get; set; }

    public string Phone { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string Gender { get; set; } = string.Empty;

    public DateOnly? DateOfBirth { get; set; }

    public string Description { get; set; } = string.Empty;

    public string AvatarUrl { get; set; } = string.Empty;

    public string RoomNumber { get; set; } = string.Empty;

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }
}
